zip-0320.rst

```diff
@@ -5,7 +5,357 @@
+   must be able to generate a receiving address such that only transparent
+   funds will be spent in transactions with an output to this address.
+2. Wallets and other Consumers that have not been upgraded to recognize the new
+   address format cannot mistake the address for another address type or
+   inadvertently send shielded funds to the address.
+3. No changes to Recipient infrastructure beyond changes to address encoding
+   and decoding should be required as a consequence of this ZIP.
+
+Alternative 1
+=============
+
+This alternative was suggested by @hanh in [#hanh-suggestion]_.
+
+TEX Addresses
+-------------
+
+A TEX Address is a Bech32m [#bip-0350]_ reencoding of a transparent Zcash
+P2PKH address [#protocol-transparentaddrencoding]_.
+
+Motivations for Alternative 1
+-----------------------------
+
+The TEX Address is the simplest possible approach to creating a new address
+type that indicates that only transparent sources of funds should be used.
+
+Specification (Alternative 1)
+-----------------------------
+
+A TEX address is produced from a Mainnet Zcash P2PKH Address by executing the
+following steps:
+
+1. Decode the address to a byte sequence using the Base58Check decoding
+   algorithm [#Base58Check]_.
+2. If the length of the resulting byte sequence is not 22 bytes or if its two-byte
+   address prefix is not :math:`[\mathtt{0x1C}, \mathtt{0xB8}]`, return an error. Otherwise,
+   let the **validating key hash** be the remaining 20 bytes of the sequence after
+   removing the two-byte address prefix.
+3. Reencode the 20-byte **validating key hash** using the Bech32m encoding
+   defined in [#bip-0350]_ with the human-readable prefix (HRP) ``"tex"``.
+
+For Testnet addresses, the required lead bytes of a P2PKH address in step 2 are
+:math:`[\mathtt{0x1D}, \mathtt{0x25}]`, and the ``"textest"`` HRP is used when
+reencoding in step 3.
+
+Wallets and other Senders sending to a TEX address (as any output) MUST
+ensure that only transparent UTXOs are spent in the creation of a
+transaction.
+
+Reference Implementation (Alternative 1)
+----------------------------------------
+
+Javascript::
+
+   import bs58check from 'bs58check'
+   import {bech32m} from 'bech32'
+   
+   // From t1 to tex
+   var b58decoded = bs58check.decode('t1VmmGiyjVNeCjxDZzg7vZmd99WyzVby9yC')
+   console.assert(b58decoded.length == 22, 'Invalid length');
+   console.assert(b58decoded[0] == 0x1C && b58decoded[1] == 0xB8, 'Invalid address prefix');
+   var pkh = b58decoded.slice(2)
+   var tex = bech32m.encode('tex', bech32m.toWords(pkh))
+   console.log(tex)
+   
+   // From tex to t1
+   var bech32decoded = bech32m.decode('tex1s2rt77ggv6q989lr49rkgzmh5slsksa9khdgte')
+   console.assert(bech32decoded.prefix == 'tex', 'Invalid address prefix')
+   var pkh2 = Uint8Array.from(bech32m.fromWords(bech32decoded.words))
+   console.assert(pkh2.length == 20, 'Invalid length');
+   var t1 = bs58check.encode(Buffer.concat([Uint8Array.from([0x1C, 0xB8]), pkh2]))
```

Comment:
Would it be possible to bypass rules for tex addresses by deriving the t1 address and sending funds there?

Reply:
Yes. However, the whole point of this idea is that if the shielded wallets used by the community enforce the constraint correctly, it would require someone to explicitly work around that constraint in order to send from a shielded address. If they do that, they're at risk of losing their funds - which is standard; if you send to an address other than the one that someone gave to you, caveat emptor.